tests/function-controller/pkg/subscription/subscription.go

```diff
@@ -0,0 +1,185 @@
+
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/util/retry"
+)
+
+const (
+	backOffJitter   = 0.1
+	backOffFactor   = 5.0
+	backOffSteps    = 10
+	backOffDuration = 4 * time.Second
+)
+
+type Subscription struct {
+	resCli      *resource.Resource
+	name        string
+	namespace   string
+	waitTimeout time.Duration
+	log         *logrus.Entry
+	verbose     bool
+}
+
+func New(name string, c shared.Container) *Subscription {
+	gvr := schema.GroupVersionResource{
+		Group:    "eventing.kyma-project.io",
+		Version:  "v1alpha1",
+		Resource: "subscriptions",
+	}
+
+	return &Subscription{
+		resCli:      resource.New(c.DynamicCli, gvr, c.Namespace, c.Log, c.Verbose),
+		name:        name,
+		namespace:   c.Namespace,
+		waitTimeout: c.WaitTimeout,
+		log:         c.Log,
+		verbose:     c.Verbose,
+	}
+}
+
+func (s *Subscription) Create(sinkUrl *url.URL) (subscription *kymaeventingv1alpha1.Subscription, err error) {
+	eventSource := &kymaeventingv1alpha1.Filter{
+		Type:     "exact",
+		Property: "source",
+		Value:    "",
+	}
+	eventType := &kymaeventingv1alpha1.Filter{
+		Type:     "exact",
+		Property: "type",
+		Value:    "sap.kyma.custom.something.order.created.v1",
+	}
+	bebFilter := &kymaeventingv1alpha1.BebFilter{
+		EventSource: eventSource,
+		EventType:   eventType,
+	}
+	bebFilters := &kymaeventingv1alpha1.BebFilters{
+		Dialect: "",
+		Filters: []*kymaeventingv1alpha1.BebFilter{
+			bebFilter,
+		},
+	}
+	subscription = &kymaeventingv1alpha1.Subscription{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Subscription",
+			APIVersion: "eventing.kyma-project.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.name,
+			Namespace: s.namespace,
+		},
+		Spec: kymaeventingv1alpha1.SubscriptionSpec{
+			Protocol:         "",
+			ProtocolSettings: &kymaeventingv1alpha1.ProtocolSettings{},
+			Sink:             sinkUrl.String(),
+			Filter:           bebFilters,
+		},
+	}
+
+	unstructuredSub, err := toUnstructured(subscription)
+	if err != nil {
+		return subscription, errors.Wrapf(err, "while creating Subscription %s", s.name)
+	}
+
+	_, err = s.resCli.Create(&unstructuredSub)
+	if err != nil {
+		return subscription, errors.Wrapf(err, "while creating Subscription %s in namespace %s", s.name, s.namespace)
+	}
+	return subscription, err
+}
+
+func (s *Subscription) Delete() error {
+	err := s.resCli.Delete(s.name)
+	if err != nil {
+		return errors.Wrapf(err, "while deleting Subscription %s in namespace %s", s.name, s.namespace)
+	}
+
+	return nil
+}
+
+func (s *Subscription) Get() (*unstructured.Unstructured, error) {
+	subscription, err := s.resCli.Get(s.name)
+	if err != nil {
+		return &unstructured.Unstructured{}, errors.Wrapf(err, "while getting Subscription %s in namespace %s", s.name, s.namespace)
+	}
+
+	return subscription, nil
+}
+
+func (s *Subscription) WaitForStatusRunning(subscription *kymaeventingv1alpha1.Subscription) error {
```

Comment:
There is no need anymore to pass the subscription to the function. For the subscription name s.Name could be used?

Reply:
Right, changed